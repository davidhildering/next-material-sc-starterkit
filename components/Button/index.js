import React from "react";
import styled from "styled-components";


const ButtonLink = styled.a`
  text-decoration: none;
`;

const ButtonLayout = styled.div`
  padding: 10px 8px;
  font-family: SFProBold, sans-serif;
  font-size: 16px;
  text-align: center;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  border: 1px solid transparent;
  color: #ffffff;
  background-color: yellowgreen;
  max-width: ${props => (props.maxWidth ? props.maxWidth : "200px")};
  border-radius: 4px;
`;

const Button = props => {
  return (
    <div>
      {props.href ? (
        <ButtonLink to={props.href}>
          <ButtonLayout maxWidth={props.maxWidth}>{props.title}</ButtonLayout>
        </ButtonLink>
      ) : (
        <ButtonLayout maxWidth={props.maxWidth}>{props.title}</ButtonLayout>
      )}
    </div>
  );
};

export default Button;
