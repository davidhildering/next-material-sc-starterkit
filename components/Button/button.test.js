import React from "react";
import { shallow } from "enzyme";
import Button from ".";

describe("Button", () => {
  it("should render correctly with no props", () => {
    const component = shallow(<Button />);

    expect(component).toMatchSnapshot();
  });
  it("should render button text correctly with given string", () => {
    const string = "kies pakket";
    const component = shallow(<Button title={string} />);
    expect(component).toMatchSnapshot();
  });
});
