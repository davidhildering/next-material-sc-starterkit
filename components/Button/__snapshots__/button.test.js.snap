// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button text correctly with given string 1`] = `
<div>
  <Button__ButtonLayout>
    kies pakket
  </Button__ButtonLayout>
</div>
`;

exports[`Button should render correctly with no props 1`] = `
<div>
  <Button__ButtonLayout />
</div>
`;
